ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/processes/OpenInTerminalAction.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.ide.processes;
+
+import static org.eclipse.che.ide.part.perspectives.project.ProjectPerspective.PROJECT_PERSPECTIVE_ID;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import java.util.Collections;
+import org.eclipse.che.ide.CoreLocalizationConstant;
+import org.eclipse.che.ide.api.action.AbstractPerspectiveAction;
+import org.eclipse.che.ide.api.action.ActionEvent;
+import org.eclipse.che.ide.api.resources.Container;
+import org.eclipse.che.ide.api.resources.Resource;
+import org.eclipse.che.ide.machine.MachineResources;
+import org.eclipse.che.ide.processes.panel.ProcessesPanelPresenter;
+import org.eclipse.che.ide.resource.Path;
+import org.eclipse.che.ide.terminal.TerminalOptionsJso;
+
+/**
+ * Action to open new terminal and navigate to selected directory.
+ *
```

Comment:
Could you please give some more tech details in the comment.
It will open the terminal window with "cd .." command?

Reply:
done